### PR TITLE
updated repartition documentation to add note about `partition_size`

### LINKED ISSUE #9276: repartition documentation doesn't specify whether the size will be reflected on output
Noticed a difference between memory usage and storage size and I think this would be a good thing to note within the documentation that partition sizes won't necessarily be reflected when storing on disk. Moving the issue here from https://github.com/geopandas/dask-geopandas/issues/210#issue-1306025315

### PR DESCRIPTION
**Pull request by drstewart19**
- [x] Closes #9276 
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -1413,11 +1413,17 @@ def repartition(
         The second-to-last item is the lower bound of the last partition, and the last
         (extra) item is the upper bound of the last partition.
 
+        Also note that `partition_size` will not necessarily be reflected on disk.
```
Comment (TomAugspurger) [inline dask/dataframe/core.py:1416]: Can you put this under the `partition_size` parameter description?

I'd phrase it as something like: "Note that the size reflects the number of bytes used as computed by :ref:`pandas.DataFrame.memory_usage`, which will not necessarily match the size on disk."

Comment (drstewart19) [reply dask/dataframe/core.py]: I went ahead and moved the clarification. I copied the same formatting, but the reference to `pandas.DataFrame.memory_usage` doesn't seem to work properly.

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -1385,7 +1385,9 @@ def repartition(
         partition_size: int or string, optional
             Max number of bytes of memory for each partition. Use numbers or
             strings like 5MB. If specified npartitions and divisions will be
-            ignored.
+            ignored. Note that the size reflects the number of bytes used as
+            computed by :ref:`pandas.DataFrame.memory_usage`, which will not
```
Comment (pavithraes) [inline dask/dataframe/core.py:1389]:
```suggestion
            computed by ``pandas.DataFrame.memory_usage``, which will not
```

File: dask/dataframe/core.py [hunk 3]
```diff
@@ -1418,6 +1420,10 @@ def repartition(
         >>> df = df.repartition(npartitions=10)  # doctest: +SKIP
         >>> df = df.repartition(divisions=[0, 5, 10, 20])  # doctest: +SKIP
         >>> df = df.repartition(freq='7d')  # doctest: +SKIP
+
+        See Also
+        --------
+        DataFrame.memory_usage_per_partition
```
Comment (pavithraes) [inline dask/dataframe/core.py:1426]:
```suggestion
        DataFrame.memory_usage_per_partition
        pandas.DataFrame.memory_usage
```

File: dask/dataframe/core.py (hunk 1 above)
Comment (pavithraes) [reply dask/dataframe/core.py]: Indeed, I don't think the `:ref:` syntax works directly with pandas docs. We can consider adding it in the "See Also" section to link to pandas docs. I'll add suggestions. :)

File: dask/dataframe/core.py (hunk 2 above)
Comment (pavithraes) [inline dask/dataframe/core.py:1389]:
```suggestion
            computed by ``pandas.DataFrame.memory_usage``, which will not
```

File: dask/dataframe/core.py (hunk 3 above)
Comment (pavithraes) [inline dask/dataframe/core.py:1426]:
```suggestion
        DataFrame.memory_usage_per_partition
        pandas.DataFrame.memory_usage
```

**Comment by pavithraes** [comment]
Looking at the RTD build for this PR, I see the `pandas.DataFrame.memory_usage` link only in [Series.repartition](https://dask--9288.org.readthedocs.build/en/9288/generated/dask.dataframe.Series.repartition.html) and not in [DataFrame.repartition](https://dask--9288.org.readthedocs.build/en/9288/generated/dask.dataframe.DataFrame.repartition.html). Is it just me?

I see it on both pages on my local docs build though.

**Pull request merged**